### PR TITLE
rbd: Enable featuregates for volume expansion recovery

### PR DESCRIPTION
**Pull request by humblec**
The enabled feature gate helps us to recover from volume expansion
failure.

Additional Reference#  

Reference below URL for this feature gate details:
https://github.com/kubernetes-csi/external-resizer/

Additional Note for reviewer:

No E2E as its a generic one and not specific to the Ceph CSI driver as like `AnnotateFSResize`. 

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml [hunk 1]
```diff
@@ -104,6 +104,8 @@ spec:
             - "--leader-election"
             - "--retry-interval-start=500ms"
             - "--handle-volume-inuse-error=false"
+            #  set it to true to try volume expansion recovery
+            - "--feature-gates=RecoverVolumeExpansionFailure=true,AnnotateFsResize=true"
```
Comment (pkalever) [inline deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: use label `deploy` in the commit-msg instead of `rbd`

Comment (humblec) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: @pkalever this is a draft PR :) , I will be rearranging the commits

Comment (pkalever) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: Yes, I'm aware that it is a draft. Just noticed it, so ... :-)

Comment (humblec) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: sure..thanks..

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM but @humblec What are the cases the volume expansion can fail in cephcsi?

**Comment by humblec** [comment]
> changes LGTM but @humblec What are the cases the volume expansion can fail in cephcsi?

it could be anything, more or less, this is a fallback mechanism for the user to come back from an error scenario.

**Comment by humblec** [comment]
@Mergifyio rebase

File: scripts/minikube.sh [hunk 2]
```diff
@@ -226,7 +226,7 @@ up)
     #  get kubernetes version we are operating on and accordingly enable feature gates
     KUBE_MAJOR=$(kube_version 1)
     KUBE_MINOR=$(kube_version 2)
-    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -ge 22 ];then
+    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -eq 22 ];then
```
Comment (Madhu-1) [inline scripts/minikube.sh]: why this change is required?

File: scripts/minikube.sh [hunk 3]
```diff
@@ -235,6 +235,10 @@ up)
         # if kubernetes version is greater than 1.22 disable PodSecurity feature gate
         K8S_FEATURE_GATES="${K8S_FEATURE_GATES},PodSecurity=false"
     fi
+    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -ge 23 ];then
+        # if kubernetes version is greater than 1.22 enable RWOP feature gate
```
Comment (Madhu-1) [inline scripts/minikube.sh]: comment is not matching the featureGate which is getting enabled.

File: deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml [hunk 4]
```diff
@@ -68,6 +68,8 @@ spec:
             - "--leader-election"
             - "--retry-interval-start=500ms"
             - "--handle-volume-inuse-error=false"
+            #  set it to true to try volume expansion recovery
```
Comment (Madhu-1) [inline deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml]: remove the comment. this is not required, flag is self explanatory

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

**Comment by humblec** [comment]
@Madhu-1 comments addressed, ptal. thanks.

**Comment by humblec** [comment]
@Madhu-1 ptal. comments are addressed

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio refresh
